packages/@aws-cdk/aws-appsync/lib/data-source.ts

```diff
@@ -363,4 +364,31 @@ export class RdsDataSource extends BackedDataSource {
       scope: this,
     });
   }
+}
+
+/**
+ * Properities for the Elasticsearch Data Source
+ */
+export interface ElasticsearchDataSourceProps extends BackedDataSourceProps {
+  /**
+   * The elasticsearch domain containing the endpoint for the data source
+   */
+  readonly domain: IDomain;
+}
+
+/**
+ * An Appsync datasource backed by Elasticsearch
+ */
+export class ElasticsearchDataSource extends BackedDataSource {
+  constructor(scope: Construct, id: string, props: ElasticsearchDataSourceProps) {
+    super(scope, id, props, {
+      type: 'AMAZON_ELASTICSEARCH',
+      elasticsearchConfig: {
+        awsRegion: props.domain.stack.region,
+        endpoint: `https://${props.domain.domainEndpoint}`,
```

Comment:
prepending https manually is required here? feels weird.

Reply:
I can't find any documentation that explicitly says this.. but this is the cloudformations deploy error I get when I don't add the `https` string interpolation

```sh
AWS::AppSync::DataSource    | api/ds (apids4328272F) Property validation failure: [Value for property {/ElasticsearchConfig/Endpoint} does not match pattern {(?i)https?://.*$}]
```

Reply:
O wait i see an issue now..

`http` and `https` protocols are both allowed bc of the regex

Reply:
hm.. can't really find a way with the current elasticsearch domain class to generate the endpoint correctly.. mostly because the `enforceHttps` prop is not member

Reply:
hmm alright, we will revisit if it becomes an issue.